pkg/k8s/watchers/pod.go

```diff
@@ -148,6 +148,13 @@ func (k *K8sWatcher) podsInit(slimClient slimclientset.Interface, asyncControlle
 		return isConnected
 	}
 
+	// Disable watching pods if we are in high-scale mode. We don't need to
+	// insert pod IPs into the ipcache.
+	if option.Config.EnableHighScaleIPcache {
+		asyncControllers.Done()
+		return
+	}
```

Comment:
One thing I realized in this patch now (:sweat_smile:) is that named ports support comes from the pod watcher because named ports are defined in the pod resource, and then referenced in CNPs. 

I see two paths:

1. Document that we don't support named ports with high-scale ipcache mode
2. Keep the pod watcher, but make the pod IP -> ipcache code path conditional on high-scale ipcache mode

Reply:
I'd rather go for option 1 and add support later if asked for. We don't have a flag to control support for named ports, right?

Reply:
As much as I am reticent to further complicate the matrix of compatible features in Cilium... I think that Option 1 is probably the reasonable approach here. I'm not convinced that the named ports policy feature was specified in a scalable way.